src/util/retry.rs

```diff
@@ -0,0 +1,226 @@
+use std::error::Error;
+use std::future::Future;
+use std::marker::PhantomData;
+use std::time::Duration;
+
+use rand::Rng;
+
+/// A function that performs and retries the given operation, according to a retry policy and a set
+/// of retriable errors.
+///
+/// **Caution**: A retry policy without the number of attempts capped by `MaxAttemptsRetryPolicy`
+/// decorator will result in infinite retries.
+///
+/// **Usage**:
+/// ```rust
+/// # use std::time::Duration;
+/// # use vss_client::error::VssError;
+/// # use vss_client::util::retry::{ExponentialBackoffRetryPolicy, retry, RetryPolicy};
+/// #
+/// # let base_delay = Duration::from_millis(10);
+/// # 		let max_attempts = 3;
+/// # 		let max_total_delay = Duration::from_secs(60);
+/// # 		let max_jitter = Duration::from_millis(5);
+/// #
+/// # async fn operation() -> Result<i32, VssError>  {
+/// # 	tokio::time::sleep(Duration::from_millis(10)).await;
+/// # 	Ok(42)
+/// # }
+/// #
+/// let retry_policy = ExponentialBackoffRetryPolicy::new(base_delay)
+/// 		.with_max_attempts(max_attempts)
+/// 		.with_max_total_delay(max_total_delay)
+/// 		.with_max_jitter(max_jitter)
+/// 		.skip_retry_on_error(|e| matches!(e, VssError::InvalidRequestError(..)));
```

Comment:
Only one indentation required. For docs, I've tended to just use four spaces per indent, excluding the space after `///`, but we haven't been consistent about it.